code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -80,9 +80,12 @@
 			user << "<span class='danger'>[src] buzzes and beeps.</span>"
 
 /mob/living/simple_animal/bot/cleanbot/process_scan(obj/effect/decal/cleanable/D)
+	. = list()
 	for(var/T in target_types)
 		if(istype(D, T))
-			return D
+			. += D
+	if(.)

```

Comment:
I think byond handles that by returning false (out of bounds as false, that is) which is perfectly fine for us, but I will do a test.